src/sgame/sg_bot.cpp

```diff
@@ -263,6 +262,188 @@ bool G_BotSetDefaults( int clientNum, team_t team, int skill, Str::StringRef beh
 	return true;
 }
 
+struct botSkillTreeElement_t {
+	Str::StringRef name;
+	enum bot_skill enable;
+	int cost;
+	// function that determines if this skill is available
+	std::function<bool(const gentity_t *self, skillSet_t existing_skills)> predicate;
+	// skills that are unlocked once you unlock this one
+	std::vector<botSkillTreeElement_t> unlocked_skills;
```

Comment:
What does this mean? Total cost of this skill = cost + sum(unlocked_skills.cost)?

Reply:
Ah I see now. It's more of a skill tree. You need the parent skill before you can have the "unlocked" skills.